src/inmanta/deploy/scheduler.py

```diff
@@ -490,9 +508,13 @@ async def _new_version(
                     elif details.attribute_hash != self._state.resources[resource].attribute_hash:
                         # The desired state has changed.
                         new_desired_state.add(resource)
+                    # Nothing need to be done for these resources' timers yet. They will be
+                    # picked up by the scheduler eventually, and only then, after deployment
+                    # action should be taken wrt timers.
                 else:
                     # It's a resource we don't know yet.
                     new_desired_state.add(resource)
+                    self._timer_manager.update_resource(resource, dirty=True)
```

Comment:
I don't think this is right: we're scheduling the resource *now*, so we don't need a timer for later.